### PR TITLE
[4.0] [workflow] - notice on edit plugin

### PR DESCRIPTION
**Pull request by alikon**
Pull Request for Issue #21599 .

### Summary of Changes
removed Undefined property: Joomla\\CMS\\Object\\CMSObject::$id in /var/www/html/j4dev/layouts/joomla/edit/global.php

### Testing Instructions
edit a plugin

### CODE REVIEW

**Comment by ghost** [comment]
I have tested this item :white_check_mark: successfully on 57542fa8eea2e3ebf087be044edcf210fd62dfc0<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/21623">issues.joomla.org/tracker/joomla-cms/21623</a>.</sub>

**Comment by infograf768** [comment]
I had seen this error indeed before, but I am wondering concerning taking off these lines of code.
Are they necessary in some cases?

**Comment by alikon** [comment]
for what can i undestand NO, but consider that i'm still trying to understand workflow :face_with_head_bandage:

**Comment by infograf768** [comment]
I did a grep for `<?php echo JLayoutHelper::render('joomla.edit.global', $this); ?>`
and it is used elsewhere than plugin edit without error (15 times in present 4.0).

Therefore I think we have to dig further.

**Comment by infograf768** [comment]
@bembelimen 
What about this?

**Comment by alikon** [comment]
@infograf768 i've made it a more conservative fix

File: layouts/joomla/edit/global.php [hunk 1]
```diff
@@ -40,11 +40,10 @@
 	'version_note',
 );
 
-if ($displayData->get('item')->id !== null)
+if (isset($displayData->get('item')->id) && ($displayData->get('item')->id !== null))
 {
 	array_unshift($fields, 'transition');
 }
-
```
Comment (infograf768) [inline layouts/joomla/edit/global.php]: any reason to kill that empty line?

Comment (alikon) [reply layouts/joomla/edit/global.php]: no, simply lazy :smile:

**Comment by infograf768** [comment]
I have tested this item :white_check_mark: successfully on d0211c2321f8c6cdc1bd3b5e6714833f063e3e2d<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/21623">issues.joomla.org/tracker/joomla-cms/21623</a>.</sub>

**Comment by franz-wohlkoenig** [comment]
Ready to Commit after two successful tests.

**Comment by ghost** [comment]
I have tested this item :white_check_mark: successfully on d0211c2321f8c6cdc1bd3b5e6714833f063e3e2d<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/21623">issues.joomla.org/tracker/joomla-cms/21623</a>.</sub>

**Comment by ghost** [comment]
Ready to Commit after two successful tests.

**Comment by wilsonge** [comment]
Before merging this I just want to hear from @bembelimen exactly what this code is doing

**Comment by bembelimen** [comment]
To be honest, I have to guess, why it's implemented that way, because it was from the student. But what it does is: it makes sure, that the transitions will only be displayed when the "item" is not new. (New items have no transiton)

I think, this check should be implemented differently: The component should decide, if the field is visible or not, not the layout based on the ID.

**Comment by brianteeman** [comment]
its a bit of a problem if no one knows why the code is written this way - surprised

**Comment by infograf768** [comment]
Anyone to propose another patch?

**Comment by bembelimen** [comment]
> its a bit of a problem if no one knows why the code is written this way - surprised

Because the developer decided to do it that way like all the code in Joomla! is written because someone decided to do it in a way or another.

It's important to know why the code is here => see my answer above...

**Comment by wilsonge** [comment]
OK Merging this so we loose the notice. But can someone open an issue to actually review what's going on here. I guess this is so on first save of the article you can't immediately transition. But that may not be a bad thing...

**Comment by ghost** [comment]
@wilsonge will open an Issue.

**Pull request merged**